post-form.js

```diff
@@ -0,0 +1,96 @@
+import {parseFilename} from './lib.js';
+
+export default class CreatePost extends HTMLElement {
+  constructor() {
+    super();
+    this._root = this.attachShadow({mode: 'open'})
+    this._root.innerHTML = `
+      <link href="index.css" rel="stylesheet">
+      <style>
+        div.formContainer {
+          display: flex;
+          flex-direction: column;
+          gap: 5px;
+          margin: 1em 0;
+        }
+      </style>
+      <form id="postForm"></form>
+      <div class="formContainer">
+        <label>Title</label>
+        <input name="title" form="postForm"/>
+        <label>Body</label>
+        <file-upload></file-upload>
+        <textarea name="content" rows="15" form="postForm"></textarea>
+        <input type="submit" value="Create" form="postForm"/>
+      </div>
+    `
+    this._form = this._root.querySelector('#postForm');
+  }
+
+  connectedCallback() {
+    this._root.querySelector('file-upload').addEventListener('onFileSelected', (e) => {
+      this.dispatchEvent(new CustomEvent('onFileUpload', { detail: e.detail }));
+    });
+
+    this._form.addEventListener('submit', this._onSubmitForm.bind(this));
+  }
+
+  reset() {
+    this._form.reset();
+  }
+
+  _onSubmitForm(e) {
+    e.preventDefault();
+    const originalFilename = this.getAttribute('filename') || null;
+    let filename = this._root.querySelector('input[name="title"]').value;
+
+    if (! filename.match(/\d\d\d\d-\d\d-\d\d-/)){
+      let date = new Date().toISOString().slice(0, 10); //YYYY-mm-dd
+      if (originalFilename) {
+        date = parseFilename(originalFilename).date;
+      }
+      filename = `${date}-${filename}`;
```

Comment:
Should the filename be getting parsed to clean up any invalid characters beforehand?

Reply:
parse in this sense refers to extracting the date and title from the filename. As mentioned above, we should chat about how we're storing the title (and maybe the date). 

We could use frontmatter in the file contents. Or store the title as the first heading in the content.